src/signature.rs

```diff
@@ -372,23 +373,23 @@ pub fn sign(
         Algorithm::EcDsaP256(DigestAlgorithm::Sha256) => {
             let (private_key, nonce) = ecdsa_p256_sign_prep(private_key, rng)?;
             ecdsa_p256_sign_post(
-                p256::ecdsa::sign_sha256(payload, private_key.as_ref(), &nonce)
+                libcrux_ecdh::p256::ecdsa::sign_sha256(payload, private_key.as_ref(), &nonce)
```

Comment:
The ECDH crate is exposing a signature API? That doesn't sound right.
The code is all mixed up in HACL, but we should have separate crates for them or keep the signature parts in the top level here for now.

Reply:
Yeah, you're right, that doesn't make sense. I've split off that part of the HACL API and moved it back to libcrux. Should we expose it there directly or via re-export from `libcrux::ecdh`, as before?